### ISSUE TITLE
Too much Recursion 

### ISSUE THREAD

**Issue by mes258** [vinport16/chunkball#27]
<img width="666" alt="image" src="https://user-images.githubusercontent.com/20585567/211657990-238d574d-6757-4d2c-b6a7-f86ca95ce978.png">

Game lagged after 2 minutes with this error. Map size was 120 x 60 x 120

**Comment by vinport16** [comment]
this seems to happen when you're moving fast and jumping up into a wall or ceiling. but kind of hard to reproduce on purpose

**Comment by vinport16** [comment]
fixed in 18c21f69cedb39efb51e5e026912d230e4ab811e
I added a tiny bit of friction between iterations of the nextPosition function. Unfortunately I'm not sure what exactly was causing this issue, but I was able to exacerbate it by changing the number of iterations in this loop:
```
// halve the distance 7 times until you find a position that is not colliding
for (let i = 0; i < 7; i++) {
```
to 2, which makes the issue basically unavoidable. Even in this extremely bad case, adding just 0.01 friction prevented the stack overflow.
